contracts/core/RebalancingSet.sol

```diff
@@ -0,0 +1,389 @@
+/*
+    Copyright 2018 Set Labs Inc.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+*/
+
+pragma solidity 0.4.24;
+
+import { DetailedERC20 } from "zeppelin-solidity/contracts/token/ERC20/DetailedERC20.sol";
+import { SafeMath } from "zeppelin-solidity/contracts/math/SafeMath.sol";
+import { StandardToken } from "zeppelin-solidity/contracts/token/ERC20/StandardToken.sol";
+import { ISetFactory } from "./interfaces/ISetFactory.sol";
+
+/**
+ * @title SetToken
+ * @author Set Protocol
+ *
+ * Implementation of Rebalancing Set token.
+ */
+
+
+contract RebalancingSet is
+    StandardToken,
+    DetailedERC20
+{
+    using SafeMath for uint256;
+
+    /* ============ Enums ============ */
+
+    enum State { Default, Proposal, Rebalance }
+
+    /* ============ State Variables ============ */
+
+    address public factory;
+    uint256 public naturalUnit = 1;
+    address public manager;
+    State public rebalanceState;
+
+    // State updated after every rebalance
+    address public currentSet;
+    uint256 public shareRatio;
+    uint256 public lastRebalanceTimestamp;
+
+    // State governing rebalance cycle
+    uint256 public proposalPeriod;
+    uint256 public rebalanceInterval;
+
+    // State to track proposal period
+    uint256 public proposalStartTime;
+
+    // State needed for auction/rebalance
+    uint256 public auctionStartTime;
+    address public rebalancingSet;
+    address public auctionLibrary;
+    uint256 public auctionPriceDivisor;
+    uint256 public auctionStartPrice;
+    uint256 public curveCoefficient;
+    address[] public combinedTokenArray;
+    uint256[] public combinedCurrentUnits;
+    uint256[] public combinedRebalanceUnits;
+    uint256 public remainingCurrentSets;
+    uint256 public rebalanceSetSupply;
+
+    /* ============ Events ============ */
+
+    event NewManagerAdded(
+        address newManager,
+        address oldManager
+    );
+
+    event RebalanceProposed(
+        address rebalancingSet,
+        address indexed auctionLibrary,
+        uint256 indexed proposalPeriodEndTime
+    );
+
+    event RebalanceStarted(
+        address oldSet,
+        address newSet
+    );
+
+    event RebalanceFinished(
+        address newSet,
+        uint256 newShareRatio
+    );
+
+
+    /* ============ Constructor ============ */
+
+    /**
+     * Constructor function for Rebalancing Set Token
```

Comment:
Perhaps we note that this initial Rebalancing Set uses a manager vs. other oracle system

Reply:
In my mind this can use either. It's just an address, all someone needs to do is prop up a smart contract that acts as the oracle and set that as the manager.